src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -200,11 +200,10 @@ public override void OnException(Activity activity, object payload)
                         {
                             case SocketError.HostNotFound:
                                 activity.SetStatus(Status.Error.WithDescription(exc.Message));
-                                return;
+                                break;
                         }
                     }
-
-                    if (exc.InnerException != null)
+                    else if (exc.InnerException != null)
```

Comment:
The flow seems a bit weird. Consider moving the `exc.InnerException is SocketException exception` check inside `exc.InnerException != null`?

Reply:
yup. 
Also there is no need of a Switch case... as there is a single check..

Reply:
So I moved the check, because I agree the `else if` was wonky. Though, yes as the code currently stands, there's zero need to explicitly check for the SocketException. It seems there was once some thought that we'd set the status description differently based on the type of exception, but clearly that was not actually happening in practice. So I guess I don't know what the original intent was here 😆.

I'll see if the history can shed any light on this...

Reply:
Ah ha ok, all of this dates back to the time that Span.Status was more than just `OK`, `ERROR`, or `UNSET`

https://github.com/open-telemetry/opentelemetry-dotnet/blob/1ed4a289276d2bcd8f19d540faa1fc3e845280d9/src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpHandlerDiagnosticListener.cs#L161-L177

I simplified it.